yarpl/include/yarpl/observable/ObservableOperator.h

```diff
@@ -165,31 +167,31 @@ template <
     typename D,
     typename F,
     typename = typename std::enable_if<std::is_callable<F(U), D>::value>::type>
-class MapOperator : public ObservableOperator<U, D> {
+class MapOperator : public ObservableOperator<U, D, MapOperator<U, D, F>> {
+  using ThisOperatorT = MapOperator<U, D, F>;
```

Comment:
Do we need to write these two lines again and again in each sub class.
Is it possible to define them in the ObservableOperator class? 
If we do so, can we still be able to use them in the child classes?

Reply:
Unqualified names aren't looked up in the parent class, so you'd end up prefixing everything with `ObservableOperator<child class template params>::<the type>` each time